### PR TITLE
make sure response error when TOC parse failed

### PR DESCRIPTION
**Pull request by songjiayang**
Signed-off-by: songjiayang <songjiayang1@gmail.com>

### CODE REVIEW

**Comment by songjiayang** [comment]
hi @roidelapluie , thanks for your review.

I think  `d.Error()` will always be nil,  becuase the `d` is a new `encoding.Decbuf` variable.  So I want put the error check after `d.Be64()` ,  but the bs.length check  at the first,  so the `d.Error()` also is nil even put it last.

File: tsdb/index/index.go [hunk 1]
```diff
@@ -175,18 +175,14 @@ func NewTOCFromByteSlice(bs ByteSlice) (*TOC, error) {
...
-
 	return &TOC{
 		Symbols:           d.Be64(),
 		Series:            d.Be64(),
 		LabelIndices:      d.Be64(),
 		LabelIndicesTable: d.Be64(),
 		Postings:          d.Be64(),
 		PostingsTable:     d.Be64(),
-	}, nil
+	}, d.Err()
```
Comment (codesome) [inline tsdb/index/index.go:185]:
```go
	toc := &TOC{
		Symbols:           d.Be64(),
		Series:            d.Be64(),
		LabelIndices:      d.Be64(),
		LabelIndicesTable: d.Be64(),
		Postings:          d.Be64(),
		PostingsTable:     d.Be64(),
	}
	return toc, d.Err()
```

**Review by codesome — COMMENTED** [review]
Nice catch! Just to be explicit about the order and not rely on the compiler ordering, I have suggested a small modification.

**Review by roidelapluie — APPROVED** [review]

**Comment by roidelapluie** [comment]
Thanks!

**Pull request merged**